xarray/core/groupby.py

```diff
@@ -479,26 +509,47 @@ def __init__(
             If True, also restore the dimension order of multi-dimensional
             coordinates.
         """
-        self.groupers = groupers
-
         self._original_obj = obj
+        self._restore_coord_dims = restore_coord_dims
+        self.groupers = groupers
 
-        (grouper,) = self.groupers
-        self._original_group = grouper.group
+        if len(groupers) == 1:
+            (grouper,) = groupers
+            self.encoded = grouper.encoded
+        else:
+            for grouper in groupers:
+                if grouper.group.ndim > 1:
+                    raise NotImplementedError(
```

Comment:
This just needs some tests to enable.

Reply:
@max-sixty a test for this path would be good. I can fix it up if it doesn't work